### ISSUE TITLE
[Question]: Overwrite Relation Methods

### ISSUE THREAD

**Issue by johannesschobel** [doug-martin/nestjs-query#47]
Dear @doug-martin ,

i just wanted to ask, if it is possible to overwrite the relationship methods of the `Resolver` class.
Consider the following example:

I have 2 `ObjectTypes` for `User` and `Group` with their corresponding `Entitites` and have my `Resolver` set up like this:
```ts
@Resolver(of => UserObject)
export class UserResolver extends CRUDResolver(UserObject, {
  create: { many: { disabled: true }, CreateDTOClass: CreateUserInput },
  delete: { disabled: true },
  update: { disabled: true },
  relations: {
    many: {
      groups: {
        relationName: 'groups',
        DTO: GroupObject,
        nullable: true
      },
    },
  },
}) {
  constructor(
    readonly service: UserService
  ) {
    super(service);
  }
```

This will auto-generate some methods to add / update / delete the assignment between `User` and `Group`. However, in my own business logic, i first need to check, if a `User` has accepted the "terms of service" before he actually can join a `Group`.
In order to do this, i need to add custom logic to the `addUserToGroups` method.

Basically i would need to do something like this:
```ts
@Mutation(...)
async addUserToGroups(...) {
  if (user.acceptedTOS === false) {
    // throw an exception
  }
  return super.addUserToGroups(params);
}
```

How can i do this? Does this work?
All the best and thanks for your help.

**Comment by doug-martin** [comment]
Hi @johannesschobel, I would use a guard to do this, currently using guards with relations applies it to all operations related to the relation. 

Do you want to check that they have have accepted terms in both add and remove? If you want it to only appy to the add operations It will require an enhancement to specify certain guards for the operation types.

**Comment by johannesschobel** [comment]
Dear @doug-martin ,
i am not sure, if it is possible to use a guard for this approach, as i may need some more complex business logic (not sure yet).

For now, I managed to work around it like this:
```ts
@InputType()
export class SetGroupOnUserInput extends RelationInputType() {}

@Mutation()
  async setGroupOnUser(
    @Args('input') input: SetGroupOnUserInput,
  ) {
    await this.service.setRelation('group', input.id, input.relationId);
    return await this.findById(input.id);
  }
```

May not be the best solution, but works (for now :laughing: )
